### DISCUSSION TITLE
How to customize the aria2 path in Scoop?

### DISCUSSION THREAD

**Discussion by brucmao** [ScoopInstaller/Scoop#5592] [Help and Questions]
I want to use my own aria2 configuration. 
My aria2 installation path is located at `d:\Personal\totalcmd\Tools\aria2\aria2c.exe`

**Comment by rashil2000** [comment]
This needs modification in the lib/core.ps1 file, specifically here: https://github.com/ScoopInstaller/Scoop/blob/develop/lib/core.ps1#L340, to return the location of aria2c.exe based on `Get-Command`'s output. Would you like to work on it?

**Comment by brucmao** [comment]
Modifying the source code might be too difficult for me. My main goal is to use the Aria2 RPC protocol and monitor the download progress of scoop using a Chrome browser extension.

**Comment by hrxn** [comment]
Using the default config file for aria2 should work - unless it has been specifically disabled in scoop's build, but I don't think so..

The config file for aria2 is:
```
%userprofile%\.aria2\aria2.conf
```

**Reply by brucmao** [reply]
The Aria2 that comes with SCOOP does not have a configuration file called aria2.conf.

**Reply by rashil2000** [reply]
You can create one yourself

**Reply by hrxn** [reply]
@brucmao Yes, that's normal. No program like aria2 is distributed with a config file.

You always have to create the file first or copy the config file from someone else to use and make some changes.

Config file usually means: It's a location/path of a file that a program (like aria2c) checks if it is existing, and only if it exists (and is accessible) the program instructs the usual facilities to read and process that config file.
